core/src/types/operator/blocking_operator.rs

```diff
@@ -680,6 +680,49 @@ impl BlockingOperator {
         Ok(())
     }
 
+    /// Remove the path and all nested dirs and files recursively.
+    ///
+    /// # Notes
+    ///
+    /// We don't support batch delete now.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use anyhow::Result;
+    /// # use futures::io;
+    /// # use opendal::BlockingOperator;
+    /// # fn test(op: BlockingOperator) -> Result<()> {
+    /// op.remove_all("path/to/dir")?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn remove_all(&self, path: &str) -> Result<()> {
+        let meta = match self.stat(path) {
+            Ok(metadata) => metadata,
+
+            Err(e) if e.kind() == ErrorKind::NotFound => return Ok(()),
+
+            Err(e) => return Err(e),
+        };
+
+        if meta.mode() != EntryMode::DIR {
+            return self.delete(path);
+        }
+
+        let mut obs = self.scan(path)?;
+
+        let _ = obs.try_for_each(|v| match v {
+            Ok(entry) => self.delete(entry.path()),
```

Comment:
What if there is another directory under this directory

Reply:
scan makes sure the file returns before dir.